server/chat-plugins/chatlog.ts

```diff
@@ -383,4 +602,35 @@ export const commands: ChatCommands = {
 		const roomid = targetRoom ? targetRoom.roomid : target;
 		this.parse(`/join view-chatlog-${roomid}--today`);
 	},
+
+	sl: 'searchlogs',
+	searchlog: 'searchlogs',
+	searchlogs(target, room) {
+		target = target.trim();
+		const [search, tarRoom, limit, date] = target.split(',');
+		if (!target) return this.parse('/help searchlogs');
+		if (!search) return this.errorReply('Specify a query to search the logs for.');
+		let limitString = '';
+		if (/^[0-9]+$/.test(limit)) {
+			limitString = `--limit-${limit}`;
+		} else if (toID(limit) === 'all') {
+			limitString = `--limit-all`;
+		} else if (!limit) {
+			limitString = `--limit-500`;
```

Comment:
```suggestion
		let limitString;
		if (/^[0-9]+$/.test(limit)) {
			limitString = `--limit-${limit}`;
		} else if (toID(limit) === 'all') {
			limitString = `--limit-all`;
		} else if (!limit) {
			limitString = ``;
```

This just makes the URL be shorter if you don't already have a limit.